fabtests/ubertest/uber.c

```diff
@@ -219,11 +219,17 @@ static void ft_show_test_info(void)
 static int ft_check_info(struct fi_info *hints, struct fi_info *info)
 {
 	if (info->mode & ~hints->mode) {
-		fprintf(stderr, "fi_getinfo unsupported mode returned\n");
+		fprintf(stderr, "fi_getinfo unsupported mode returned: "
+			"hints mode: %s info mode: %s\n",
+			fi_tostr(&hints->mode, FI_TYPE_MODE),
+			fi_tostr(&info->mode, FI_TYPE_MODE));
```

Comment:
fi_tostr uses an internal buffer that gets overwritten with each call.  Have you tested if this works?  You may need to move the fi_tostr() calls to separate printf calls.

Reply:
Done.